src/ModelContextProtocol/Diagnostics.cs

```diff
@@ -34,4 +37,79 @@ internal static Histogram<double> CreateDurationHistogram(string name, string de
         HistogramBucketBoundaries = [0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30, 60, 120, 300],
     };
 #endif
+
+    internal static ActivityContext ExtractActivityContext(this DistributedContextPropagator propagator, IJsonRpcMessage message)
+    {
+        string? traceparent = null;
+        string? tracestate = null;
+        propagator?.ExtractTraceIdAndState(message, ExtractContext, out traceparent, out tracestate);
```

Comment:
propagator will never be null here, right? Assuming not, we can make this:
```suggestion
        propagator.ExtractTraceIdAndState(message, ExtractContext, out var traceparent, out var tracestate);
```